modules/executable/01_mod.mk

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+exe_platforms := all
+
+ifndef bin_dir
+$(error bin_dir is not set)
+endif
+
+ifndef build_names
+ifndef exe_build_names
+$(error build_names and exe_build_names are not set)
+endif
+endif
+
+all_exe_build_names := $(sort $(build_names) $(exe_build_names))
+
+fatal_if_undefined = $(if $(findstring undefined,$(origin $1)),$(error $1 is not set))
+
+define check_variables
+$(call fatal_if_undefined,go_$1_ldflags)
+$(call fatal_if_undefined,go_$1_source_path)
+endef
+
+$(foreach build_name,$(all_exe_build_names),$(eval $(call check_variables,$(build_name))))
+
+ifdef exe_build_names
+$(call fatal_if_undefined,gorelease_file)
+endif
+
+##########################################
+
+RELEASE_DRYRUN ?= false
+
+CGO_ENABLED ?= 0
+
+run_targets := $(all_exe_build_names:%=run-%)
+build_targets := $(all_exe_build_names:%=$(bin_dir)/bin/%)
+
+$(bin_dir)/bin:
+	mkdir -p $@
+
+.PHONY: $(run_targets)
+ARGS ?= # default empty
+## Directly run the go source locally.
+## @category [shared] Build
+$(run_targets): run-%: | $(NEEDS_GO)
+	CGO_ENABLED=$(CGO_ENABLED) \
+	$(GO) run \
+		-ldflags '$(go_$*_ldflags)' \
+		$(go_$*_source_path) $(ARGS)
```

Comment:
suggestion: I was actually gonna propose removing this from the oci-image module. I don't really think there's a huge amount of value in it - better IMO to build + run the binary so you're testing something closer to what you'd release. Also prevents us having to keep things up to date in the future (i.e. making sure the go build and go run have the same GOEXPERIMENT envvars, etc)

What do you think?